_pytest/terminal.py

```diff
@@ -216,6 +217,11 @@ def write_fspath_result(self, nodeid, res):
             fspath = self.startdir.bestrelpath(fspath)
             self._tw.line()
             self._tw.write(fspath + " ")
+            width = self._tw.fullwidth - self._tw.chars_on_current_line
+            if 0 < width < 10 or self._tw.chars_on_current_line % self._screen_width > self._screen_width - 10:
```

Comment:
Why `10`? I suggest using a variable with a descriptive name instead. The same can be done for the entire conditional, for example:

```python
something_descriptive =  0 < width < 10 or self._tw.chars_on_current_line % self._screen_width > self._screen_width - 10
```

Also can you explain this logic here?

Reply:
10 symbols because of this line:
```.py [ 80%]```
this is magic 10 symbols :)

logic:
if you have long name and remaining wight less then 10 symbols, then add new line.
OR
if you have long name (3 lines) and width of tail of this name more then screen width - 10 symbols, then also add new line

Reply:
> logic:
if you have long name and remaining widht less then 10 symbols, then add new line.
OR
if you have long name (3 lines) and width of tail of this name more then screen width - 10 symbols, then also add new line 

Could you please separate each condition in their own variable with descriptive name? A descriptive variable name often is enough to explain a complex situation and avoid having to write comments or explain it to people.

Reply:
> How I can write test on it? Does it have a value like
tempdir.runtests(--screen-width=120)
?

I don't think so, but I believe `testdir` uses a hard coded value internally. You can try to match against the entire line instead of using glob wildcards.

Reply:
```
I believe testdir uses a hard coded value internally.
```
I tried to run my test and it failed because this value was changed and my test with long name on 3 lines become to test with 1 line. So that's why I deleted it.

Also, could you please some test on your terminal after my fix and before merging? I'm some afraid of my magic constants :) It looks nice in my terminals (I tried 2 terminals), but who knows how it will not on my machine.

Reply:
Hey @feuillemorte, did you push more commits? I don't see them here.

Reply:
@nicoddemus I fixed your comments, but I can't push because I can't to write a test. I try to get an object of terminal and get screen width. It says me 80 symbols, but actually it's wrong value inside runpytest() function (159 symbols):
(sorry, I can't paste image because some amazon ip's blocked in Russia, please, look url)
```
test_01_very_long_name_very_long_name_very_long_name_very_long_name_very_long_name_very_long_name_very_l="""
                import pytest
                from _pytest.terminal import TerminalReporter
                term = TerminalReporter(pytest.config)
                assert term._screen_width == 1
                
                @pytest.mark.parametrize('i', range(10))
                def test_bar(i): assert 1
            """,
```
```
test_01_very_long_name_very_long_name_very_long_name_very_long_name_very_long_name_very_long_name_very_l.py:4: in <module>
    assert term._screen_width == 1
E   assert 80 == 1
```
output:
https://pp.userapi.com/c824503/v824503903/11fc2b/8XBN-QcJDh0.jpg

Reply:
@nicoddemus I pushed some code but without tests